src/main/java/com/example/hibernate/service/ModelService.java

```diff
@@ -31,14 +32,14 @@ public Model getModelById(UUID id) throws Exception{
             return optModel.get();
     }
 
-    public Model addModel(Model model)
+    public Model addModel(final AddModel addModel)
     {
-        return modelRepository.save(model);
+        return modelRepository.save(new Model(addModel.name(), addModel.parts()));
     }
 
     public Model updateModel(@PathVariable("id") UUID id, @RequestBody Model model){
-        if (model.getId() == id) {
-            Model modelPom= modelRepository.getById(id);
+        Model modelPom= modelRepository.getById(id);
```

Comment:
Format code, ALWAYS.
Also, what if model does not exist? You need to return optional from repo, and throw exception here.